### PR TITLE
1704/sw/story 5174 anonymous training model

### PR DESCRIPTION
**Pull request by chrisatanasian**
@ColinDKelley @primiti Here are the changes to allow method overloading of setting an aggregate attribute. Please CR.

### CODE REVIEW

File: lib/aggregate/aggregate_store.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ module AggregateStore
 
     module ClassMethods
       def aggregate_attribute(name, class_name, options = {})
-        aggregated_attributes << (agg_attribute = Aggregate::AttributeHandler.factory(name, class_name, options))
+        aggregated_attribute_handlers[name] = (agg_attribute = Aggregate::AttributeHandler.factory(name, class_name, options))
```
Comment (ColinDKelley) [inline lib/aggregate/aggregate_store.rb]: Nit: I don't think the parens around (agg_attribute...) are adding anything.

File: lib/aggregate/aggregate_store.rb [hunk 2]
```diff
@@ -15,15 +15,15 @@ def aggregate_attribute(name, class_name, options = {})
       end
 
       def aggregate_has_many(name, class_name, options = {})
-        aggregated_attributes << (agg_attribute = Aggregate::AttributeHandler.has_many_factory(name, class_name, options))
+        aggregated_attribute_handlers[name] = (agg_attribute = Aggregate::AttributeHandler.has_many_factory(name, class_name, options))
```
Comment (ColinDKelley) [inline lib/aggregate/aggregate_store.rb]: Ditto.

File: lib/aggregate/aggregate_store.rb [hunk 3]
```diff
@@ -15,15 +15,15 @@ def aggregate_attribute(name, class_name, options = {})
...
 
         define_method(name)                       { load_aggregate_attribute(agg_attribute) }
         define_method("#{name}=")                 { |value| save_aggregate_attribute(agg_attribute, value) }
         define_method("#{name}_changed?")         { aggregate_attribute_changed?(agg_attribute) }
       end
 
       def aggregate_belongs_to(name, options = {})
-        aggregated_attributes << (agg_attribute = Aggregate::AttributeHandler.belongs_to_factory("#{name}_id", options))
+        aggregated_attribute_handlers[name] = (agg_attribute = Aggregate::AttributeHandler.belongs_to_factory("#{name}_id", options))
```
Comment (ColinDKelley) [inline lib/aggregate/aggregate_store.rb]: Ditto.

File: lib/aggregate/aggregate_store.rb [hunk 4]
```diff
@@ -33,7 +33,7 @@ def aggregate_belongs_to(name, options = {})
       end
 
       def aggregate_schema_version(version_number, update_callback)
-        aggregated_attributes << (agg_attribute = Aggregate::Attribute::SchemaVersion.new(version_number, update_callback))
+        aggregated_attribute_handlers[name] = (agg_attribute = Aggregate::Attribute::SchemaVersion.new(version_number, update_callback))
```
Comment (ColinDKelley) [inline lib/aggregate/aggregate_store.rb]: Ditto.

File: lib/aggregate/aggregate_store.rb [hunk 5]
```diff
@@ -114,6 +115,15 @@ def reload
       result
     end
 
+    def aggregate_attribute(name)
+      aggregate_attributes[name]
+    end
+
+    def set_aggregate_attribute(name, value)
+      agg_attribute = self.class.aggregated_attribute_handlers[name.to_sym]
```
Comment (ColinDKelley) [inline lib/aggregate/aggregate_store.rb]: This is `agg_attribute_handler` now.

File: lib/aggregate/aggregate_store.rb [hunk 6]
```diff
@@ -114,6 +115,15 @@ def reload
       result
     end
 
+    def aggregate_attribute(name)
+      aggregate_attributes[name]
+    end
```
Comment (ColinDKelley) [inline lib/aggregate/aggregate_store.rb]: As we just discussed, this should be symmetrical with the method below and only use the handler that it needs. You might want to put it back to start with `get_` therefore. (Sorry about that thrash!)

**Comment by chrisatanasian** [comment]
@ColinDKelley Thanks for the PR! Addressed all comments.

**Pull request merged**